src/AppInstallerRepositoryCore/PackageDependenciesValidation.cpp

```diff
@@ -0,0 +1,253 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+#pragma once
+#include "pch.h"
+#include <AppInstallerVersions.h>
+#include <winget/Manifest.h>
+#include <winget/ManifestValidation.h>
+#include <SQLiteWrapper.h>
+#include <PackageDependenciesValidation.h>
+#include <Microsoft/Schema/1_4/DependenciesTable.h>
+#include "Microsoft/Schema/1_0/ManifestTable.h"
+#include <winget/DependenciesGraph.h>
+
+namespace AppInstaller::Repository
+{
+    using namespace Microsoft::Schema::V1_4;
+    using namespace Microsoft::Schema::V1_0;
+
+    namespace
+    {
+        std::vector<AppInstaller::Manifest::Dependency> GetDependencies(
+            const Manifest::Manifest& manifest, AppInstaller::Manifest::DependencyType dependencyType)
+        {
+            std::vector<AppInstaller::Manifest::Dependency>  dependencies;
+
+            for (const auto& installer : manifest.Installers)
+            {
+                installer.Dependencies.ApplyToAll([&](AppInstaller::Manifest::Dependency dependency)
+                    {
+                        if (dependency.Type == dependencyType)
+                        {
+                            dependencies.emplace_back(dependency);
+                        }
+                    });
+            }
+
+            return dependencies;
+        }
+
+        std::optional<std::pair<SQLite::rowid_t, Utility::Version>> GetPackageLatestVersion(
+            SQLiteIndex* index, Manifest::string_t packageId, std::set<Utility::Version> exclusions = {})
+        {
+            SearchRequest request;
+            request.Filters.emplace_back(PackageMatchField::Id, MatchType::Exact, packageId);
```

Comment:
>Exact [](http://example.com/codeflow?start=75&length=5)

`CaseInsensitive` unless you know that every caller is getting the string from the index to begin with.